net/tsdial/peerapi_macios_ext.go

```diff
@@ -38,3 +38,15 @@ func peerDialControlFuncNetworkExtension(d *Dialer) func(network, address string
 		return lc.Control(network, address, c)
 	}
 }
+
+func interfaceIndexLocked(d *Dialer) (index int, ok bool) {
```

Comment:
This is a nice little refactor, thanks. I wonder if there's a check somewhere for "call method of value passing other fields from the same value as arguments" 😆